src/ModelContextProtocol/Protocol/Transport/HttpListenerSseServerTransport.cs

```diff
@@ -39,81 +44,69 @@ public HttpListenerSseServerTransport(McpServerOptions serverOptions, int port,
     /// <param name="port">The port to listen on.</param>
     /// <param name="loggerFactory">A logger factory for creating loggers.</param>
     public HttpListenerSseServerTransport(string serverName, int port, ILoggerFactory loggerFactory)
-        : base(loggerFactory)
     {
+        Throw.IfNull(serverName);
+
         _serverName = serverName;
+        _loggerFactory = loggerFactory;
         _logger = loggerFactory.CreateLogger<HttpListenerSseServerTransport>();
         _httpServerProvider = new HttpListenerServerProvider(port)
         {
             OnSseConnectionAsync = OnSseConnectionAsync,
             OnMessageAsync = OnMessageAsync,
         };
-    }
 
-    /// <inheritdoc/>
-    public Task StartListeningAsync(CancellationToken cancellationToken = default)
-    {
-        return _httpServerProvider.StartAsync(cancellationToken);
+        // Until we support session IDs, there's no way to support more than one concurrent session.
+        // Any new SSE connection overwrites the old session and any new /messages go to the new session.
```

Comment:
Well that seems bad :)

Reply:
Indeed. I figured it was worth calling out, but it's the current, hopefully very temporary, state of things.